classes/local/importvideosmanager.php

```diff
@@ -594,4 +615,258 @@ public static function change_acl($ocinstanceid, $sourcecourseid, $sourcecourses
         // Finall, we return the object containing the information about all 3 steps in this method.
         return $aclchangeresult;
     }
+
+    /**
+     * Gets the import mapping records of series and perform the lookup and fix.
+     * It also cleans up after the attemp is completed.
+     * @uses ltimodulemanager::fix_imported_series_modules_in_new_course() to lookup and fix series LTI modules.
+     * @uses activitymodulemanager::fix_imported_series_modules_in_new_course() to lookup and fix series Activity modules.
+     *
+     * @param int $ocinstanceid Opencast instance id.
+     * @param int $courseid Course id.
+     * @param string $newseriesid The newly created series id in the new course.
+     * @param string $restoreuid The restore unique id.
+     *
+     * @return void
+     */
+    public static function fix_imported_series_modules_in_new_course($ocinstanceid, $courseid, $newseriesid, $restoreuid) {
+
+        $seriesimportmapping = self::get_import_mapping_records([
+            'restoreuid' => $restoreuid,
+            'ocinstanceid' => $ocinstanceid,
+            'type' => self::MAPPING_TYPE_SERIES,
+            'targetcourseid' => $courseid,
+        ]);
+
+        foreach ($seriesimportmapping as $mapping) {
+            // LTI Modules.
+            ltimodulemanager::fix_imported_series_modules_in_new_course(
+                $ocinstanceid, $courseid, $mapping->sourceseriesid, $newseriesid
+            );
+
+            // Activity modules.
+            activitymodulemanager::fix_imported_series_modules_in_new_course(
+                $ocinstanceid, $courseid, $mapping->sourceseriesid, $newseriesid
+            );
```

Comment:
I though that has been taken care of in the actual method activitymodulemanager::fix_imported_series_modules_in_new_course line 653 , where we check if the return value of DB->get_records() is not empty!

Reply:
The `$DB->get_records`-call uses `opencasttype`, which is defined in `mod_opencast`, but even if this wasn't an issue, it would throw an sql exception because the `opencast` table does not exist without `mod_opencast`, so sadly it needs to be checked manually, for example with `core_plugin_manager::instance()->get_plugin_info('mod_opencast') != null`